### PR TITLE
Emit events when handling dht requests. Should fix #141

### LINKED ISSUE #141: Event for FIND-VALUE Requests
In bittorrent-dht I see there is option to specify host, if set host = true. Will it behave as real DHT node? Will it get FIND-VALUE requests from other nodes ? Is there any event to listen when node gets FIND-VALUE request from other node?

Sorry for opening an issue for question.

### PR DESCRIPTION
**Pull request by mafintosh**
@feross thoughts / concerns?

### CODE REVIEW

File: client.js [hunk 1]
```diff
@@ -432,6 +436,8 @@ DHT.prototype._onannouncepeer = function (query, peer) {
     return this._rpc.error(peer, query, [203, 'cannot `announce_peer` with bad token'])
   }
 
+  this.emit('announce_peer', query.a.info_hash, {host: host, port: peer.port})
```
Comment (feross) [inline client.js]: Can this use `infoHash` instead of `query.a.info_hash`?

File: client.js [hunk 2]
```diff
@@ -409,6 +411,8 @@ DHT.prototype._ongetpeers = function (query, peer) {
   var infoHash = query.a.info_hash
   if (!infoHash) return this._rpc.error(peer, query, [203, '`get_peers` missing required `a.info_hash` field'])
 
+  this.emit('get_peers', query.a.info_hash)
```
Comment (feross) [inline client.js]: Can this use `infoHash` instead of `query.a.info_hash`?

File: client.js [hunk 3]
```diff
@@ -400,6 +400,8 @@ DHT.prototype._onfindnode = function (query, peer) {
   var target = query.a.target
   if (!target) return this._rpc.error(peer, query, [203, '`find_node` missing required `a.target` field'])
 
+  this.emit('find_node', query.a.target)
```
Comment (feross) [inline client.js]: Can this use `target` instead of `query.a.target`?

**Comment by mafintosh** [comment]
@feross fixed. as soon as travis is green i'll merge + release this

**Pull request merged**